src/main/java/org/folio/marc4ld/service/ld2marc/resource/field/ControlFieldsBuilder.java

```diff
@@ -12,17 +12,17 @@
 import org.marc4j.marc.MarcFactory;
 
 public class ControlFieldsBuilder {
-  private static final String EMPTY_CF = SPACE.repeat(39);
+  private static final int CF_008_LENGTH = 39;
   private final Map<String, StringBuilder> storage = new LinkedHashMap<>();
 
   public ControlFieldsBuilder() {
-    storage.put(TAG_008, new StringBuilder(EMPTY_CF));
+    storage.put(TAG_008, new StringBuilder(SPACE.repeat(CF_008_LENGTH)));
   }
 
   public void addFieldValue(String tag, String newValue, Integer startPos, Integer endPos) {
     if (isNotEmpty(newValue)) {
-      storage.putIfAbsent(tag, new StringBuilder(EMPTY_CF));
-      storage.get(tag).replace(startPos, endPos, ensureLength(newValue, endPos - startPos));
+      storage.putIfAbsent(tag, new StringBuilder());
```

Comment:
Hi @pkjacob,
we have a test case you described: org.folio.marc4ld.mapper.field008.illustrations.Ld2MarcIllustrationsIT#shouldMap_onlyFour_illustrations
So it doesn't affect illustrations and I see the length is being limited here in a custom mapper: org/folio/marc4ld/service/ld2marc/mapper/custom/impl/AbstractLd2MarcBookMapper.java:30

But it still could be a problem in theory, I see your point.
So I have added a limitation to this method too, just in case.

Thank you!